docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,401 +51,388 @@ The scenarios of defining placement rules in SQL include:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=follower REPLICAS=2,
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=2;
+CREATE PLACEMENT POLICY `standardplacement1` PRIMARY REGION="us-east-1" REGIONS="us-east-1,us-east-2" FOLLOWERS=4;
+CREATE PLACEMENT POLICY `standardplacement2` LEADER_CONSTRAINTS="[+region=us-east-1]"  FOLLOWERS_CONSTRAINTS="[+region=us-east-1,+region=us-east-2]" FOLLOWERS=4;
 ```
 
-There are 4 replicas for table `test`, 2 of which are in `sh` and 2 are in `bj`.  Leader can only be placed on `sh`.
+If the constraints is specified as a dictionary (e.g. `{"+region=us-east-1":1}`) the count is not applicable and an error is returned:
 
-`ROLE` in the statement is equivalent to field `role` in PD placement rule configuration.
+```sql
+FOLLOWER_CONSTRAINTS="{+region=us-east-1:1,-region=us-east-2:2}" FOLLOWERS=3 // technically accurate, but an error
+FOLLOWER_CONSTRAINTS="{+region=us-east-1:1,-region=us-east-2:2}" FOLLOWERS=2 // an error
+```
 
-### Replicas configuration
+For dictionary format, the count is inferred by the constraint. The following constraint creates 4 followers:
 
-`REPLICAS` in the statement indicates the replica count of the specified role.
+```sql
+FOLLOWER_CONSTRAINTS="{+region=us-east-1:1,-region=us-east-2:2,+any:1}"
+```
 
-Rules defined on `leader` can omit `REPLICAS`, because the count of leader is always 1.
+Explanation:
+- 1 follower in `us-east-1`
+- 2 followers not in `us-east-2`
+- 1 follower in any region (a special label of `+any`)
 
-When all the replica counts are specified in the `CONSTRAINTS` option, `REPLICAS` can also be omitted. For example, `CONSTRAINTS="{+zone=bj:2,+zone=sh:1}", ROLE=voter` indicates that the `REPLICAS` is 3.
+`+any` changes an earlier proposal where the `FOLLOWERS` count could also be specified. This has been removed to reduce the risk of discrepancies and misconfiguration. See also "Policy Validation" below.
 
-When both `REPLICAS` and `count` in `CONSTRAINTS` are specified, it indicates that the other replicas can be placed anywhere. For example, in the case `CONSTRAINTS="{+zone=bj:2,+zone=sh:1}", ROLE=voter, REPLICAS=4`, 2 replicas are in `bj` and 1 in `sh`, and the last replica can be anywhere, including `bj` and `sh`.
+#### Built-in Placement Policies
 
-When the `CONSTRAINTS` option doesn't contain `count`, `REPLICAS` must be specified. For example, `CONSTRAINTS="[+zone=bj]" ROLE=follower` is vague, as the count of `follower` can not be inferred.
+By default every system will have two placement policies, which can be modified via `ALTER PLACEMENT POLICY` but never dropped:
 
-`REPLICAS` in the statement is equivalent to field `count` in PD placement rule configuration.
+* `default`: This policy is used only in the event that a policy has not been specified.
+* `system`: This policy is used for internal TiDB system tables.
```

Comment:
Does meta data belong to `system` too? they are structured data that are not in tables.

Reply:
Yes it should. We can also add a new builtin `metadata` if that is more appropriate.